src/altinn-app-frontend/src/components/presentation/NavBar.tsx

```diff
@@ -1,49 +1,107 @@
   showBackArrow?: boolean;
   hideCloseButton?: boolean;
+  showLanguageSelector?: boolean;
+  appLanguages: IAppLanguage[];
+  selectedAppLanguage: string;
+  onAppLanguageChange: (languageCode: string) => void;
 }
 
 const NavBar = (props: INavBarProps) => {
+  const textResources = useAppSelector(
+    (state) => state.textResources.resources,
+  );
+  const language = useAppSelector((state) => state.language.language);
+  const showLanguageSelect =
+    props.showLanguageSelector && props.appLanguages.length > 0;
+  const CloseButton = (
+    <button
+      type='button'
+      className='a-modal-close a-js-tabable-popover ml-1'
+      aria-label={getLanguageFromKey('general.close_schema', props.language)}
+      onClick={props.handleClose}
+    >
+      <span className='ai-stack'>
+        <i
+          className='ai-stack-1x ai ai-exit  a-modal-close-icon'
+          aria-hidden='true'
+        />
+      </span>
+      <span className='hidden-button-text'>
+        {getLanguageFromKey('general.close_schema', props.language)}
+      </span>
+    </button>
+  );
   return (
-    <div className='a-modal-navbar'>
-      {props.showBackArrow &&
-        <button
-          type='button'
-          className='a-modal-back a-js-tabable-popover'
-          aria-label={getLanguageFromKey('general.back', props.language)}
-          onClick={props.handleBack}
-        >
-          <span className='ai-stack'>
-            <i className='ai-stack-1x ai ai-back' aria-hidden='true' />
-          </span>
-          <span className='hidden-button-text'>
-            {getLanguageFromKey('general.back', props.language)}
-          </span>
-        </button>
-      }
-      {!props.hideCloseButton &&
-        <button
-          type='button'
-          className='a-modal-close a-js-tabable-popover'
-          aria-label={getLanguageFromKey('general.close_schema', props.language)}
-          onClick={props.handleClose}
-        >
-          <span className='ai-stack'>
-            <i className='ai-stack-1x ai ai-exit  a-modal-close-icon' aria-hidden='true' />
-          </span>
-          <span className='hidden-button-text'>
-            {getLanguageFromKey('general.close_schema', props.language)}
-          </span>
-        </button>
-      }
-    </div>
+    <Box
+      width={'100%'}
+      display='flex'
+      justifyContent={'space-between'}
+      alignItems={'flex-end'}
+      className='mt-3'
+    >
+      <div>
+        {props.showBackArrow && (
+          <button
+            type='button'
+            className='a-modal-back a-js-tabable-popover'
+            aria-label={getLanguageFromKey('general.back', props.language)}
+            onClick={props.handleBack}
+          >
+            <span className='ai-stack'>
+              <i className='ai-stack-1x ai ai-back' aria-hidden='true' />
+            </span>
+            <span className='hidden-button-text'>
+              {getLanguageFromKey('general.back', props.language)}
```

Comment:
I did a little digging and noticed that the `hidden-button-text` class above only has a `display: none` rule, making it completely irrelevant, as it's never rendered.
We are moving away form using global stylesheets and using component based styles, but for now you can switch alle the cases with `hidden-button-text` to `sr-only` as it is also defined in the old design-system and has a more screen-reader-friendly definition.